R/expr-sym.R

```diff
@@ -24,51 +24,28 @@ syms <- function(x) {
   map(x, sym)
 }
 
-#' Coerce an object to a name or call.
+#' Coerce a symbol to a name
 #'
-#' These coercing functions can transform names, calls, formulas, and
-#' strings. The distinction between a name and a call is particularly
-#' important when coercing from a string. Coercing to a call will
-#' parse the string, coercing to a name will create a (potentially)
-#' non-syntactic name.
-#'
-#' @param x An object to coerce
+#' @param x A symbol.
+#' @return A string.
+#' @seealso [sym()] for the inverse operation.
 #' @export
-#' @return `as_symbol()` and `as_lang()` return a symbol or a
-#'   call. `as_name()` returns a string.
 #' @examples
-#' as_symbol("x + y")
-#' as_lang("x + y")
+#' sym <- quote(sym)
+#' as_name(sym)
 #'
-#' as_lang(~ f)
-#' as_symbol(~ f())
-as_symbol <- function(x) {
-  coerce_type(x, "symbol",
-    symbol = x,
-    string = sym(x),
-    quosure = as_symbol(f_rhs(x)),
-    language =
-      switch_lang(x,
-        namespaced = node_car(x),
-        inlined = abort("Can't create symbol from inlined call"),
-        recursive = abort("Can't create symbol from recursive call"),
-        as_symbol(node_car(x))
-      )
-  )
-}
-#' @export
-#' @rdname as_symbol
+#' quo <- quo(sym)
+#' as_name(get_expr(quo))
 as_name <- function(x) {
-  switch_type(x,
+  coerce_type(x, "a name",
     string = x,
-    as_string(as_symbol(x))
+    symbol = as_string(x)
```

Comment:
oops, just realised that now we've removed `as_symbol()`, there is no sense in keeping `as_name()` since we have `as_string()`. I'll remove it as well.